src/performance.js

```diff
@@ -60,6 +68,49 @@ export class Performance {
     this.events_ = [];
   }
 
+  /**
+   * Listens to viewer and resource events.
+   */
+  coreServicesAvailable() {
+    this.viewer_ = viewerFor(this.win);
+    this.resources_ = resourcesFor(this.win);
+
+    this.viewer_.onVisibilityChanged(this.flush.bind(this));
+
+    let timeReqToBeVisible = 0;
+
+    // If doc didnt start in prerender mode, then no need to
+    // listen for when the first resource is laid out.
+    if (this.viewer_.didStartInPrerender()) {
+      this.resources_.whenFirstResourceLaidOut().then(() => {
+        if (this.viewer_.hasBeenVisible()) {
+          // Prerender Completed, and user has requested
+          // doc to be visible prior to prerender completion.
+          // User might have been seeing the first set of resources
+          // loading.
+          this.tick('pc', null, timer.now() - timeReqToBeVisible);
```

Comment:
@dvoytenko should i use `this.firstPassAfterDocumentReady_` as to know when to stop collecting or will the elements in the first invocation of `discoverWork` be enough?